library/aes.c

```diff
@@ -1030,6 +1030,9 @@ int mbedtls_aes_crypt_ctr( mbedtls_aes_context *ctx,
     int c, i;
     size_t n = *nc_off;
 
+    if ( n > 0x0F )
```

Comment:
Passing `n > 0x0F` is a programmer error. It's good that we detect it rather than corrupt memory, but it isn't required.

If you found this because something is passing an invalid value, that thing must be fixed.

Reply:
I understand that crypto primitives from mbed TLS can be used on their own. Hence this function isn't internal. Isn't it a good idea to do data validation in user facing API. As you said it adds robustness.